ratis-client/src/main/java/org/apache/ratis/client/api/DataStreamApi.java

```diff
@@ -37,4 +43,10 @@
 public interface DataStreamApi {
   /** Create a stream to send data. */
   DataStreamOutput stream();
+
+  /** Send data using a stream with the submessage size. */
+  CompletableFuture<DataStreamReply> streamAsync(Message message, SizeInBytes submessageSize);
```

Comment:
For DataStreamApi, the streamAsync(..) methods may not be very useful since it uses Message so that it requires buffer copying.